Utils/AutoRegistryKeyHandle.cpp

```diff
@@ -0,0 +1,45 @@
+#include "pch.h"
+#include "AutoRegistryKeyHandle.h"
+
+AutoRegistryKeyHandle::AutoRegistryKeyHandle(const HKEY& handle)
+	: m_handle(handle)
+{
+}
+
+AutoRegistryKeyHandle::~AutoRegistryKeyHandle()
+{
+	try
+	{
+		this->registryKeyHandleDeleter();
+	}
+	catch (...)
+	{
+		// Intentionally left black
```

Comment:
Add DEBUG_PRINT of generic message and we will replace it latter with logging into log file, something like: "Exception was thrown in AutoRegistryKeyHandle destructor"